### DISCUSSION TITLE
HttpClient Timeout Clarification

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#83530] [Q&A] [answered]
When does the timer for HttpClient.Timeout start and when does it stop?

Basically, I want the timeout to start after SendAsync and stop once ResponseHeaders are received.

There are a few different key points I can imagine :
HttpClient.SendAsync
DSN Resolution Complete
Socket Connection Complete
Start sending HttpRequest
Finish sending HttpRequest
Waiting for HttpResponse
Http Response Headers Received
Http Body Received

**Comment by stephentoub** [comment, marked as answer]
> When does the timer for HttpClient.Timeout start and when does it stop? Basically, I want the timeout to start after SendAsync and stop once ResponseHeaders are received.

It starts when SendAsync is called.  It stops when the task returned from SendAsync completes, so when headers are completed with HttpCompletionOption.ResponseHeadersRead or when the body is completed with HttpCompletionOption.ResponseContentRead.
